### ISSUE TITLE
Login should trigger password save promt

### ISSUE THREAD

**Issue by JaySunSyn** [convoo/login-fire#170]
After a signup or login, chrome's and other password manger should display a prompt to save the password.

This usually happens via a form submission. But can we fake it somehow?

**Comment by JaySunSyn** [comment]
Oh, maybe connected to https://github.com/convoo/login-fire/pull/145? and #31

**Comment by masonlouchart** [comment]
I don't know if it's related to the Credential API but browsers should definitively prompt for saving password. Thank you for having opened this issue.

**Comment by Oupsla** [comment]
I think the problem is the use of paper-input and the shadow-root.
Chrome (and other browsers obviously) seems to got a bug with autocomplete and the shadow root (See here : https://bugs.chromium.org/p/chromium/issues/detail?id=746593 and this issue https://github.com/PolymerElements/paper-input/issues/554)

Maybe could be better to implement the solution of the last comment on this issue  :

> you can use paper-input-container, which is a decorator for an <input>, and use that inside a form; this will look like a paper-input and will get autofilled.

**Comment by masonlouchart** [comment]
I wanted let you know I worked from while to refactor the project and I took the time today to get back in this work and to [push on a new branch](https://github.com/convoo/login-fire/tree/refactoring-for-next-release). Feel free to review before the PR and stop me before I went to far or ask questions.

**tl;dr**
- the login logic and rendering were seperated
- the progress control issue was addresed
- the duplicate events firing was fixed
- unit test suites were added

**Comment by JaySunSyn** [comment]
Awesome work @masonlouchart !!
